cartridge-cli.lua

```diff
@@ -841,18 +854,21 @@ local function detect_git_version(source_dir)
 end
 
 local function find_rockspec(source_dir)
-    for _, file in ipairs(listdir(source_dir)) do
+    local files, err = listdir(source_dir)
+    if files == nil then return nil, err end
+
+    for _, file in ipairs(files) do
         if string.endswith(file, '.rockspec') then
             return file
         end
     end
 end
 
 local function detect_name(source_dir)
-    local rockspec = find_rockspec(source_dir)
-    if rockspec ~= nil then
-        return string.match(rockspec, '^(%g+)%-scm%-1%.rockspec$')
-    end
+    local rockspec, err = find_rockspec(source_dir)
+    if rockspec == nil then return nil, err end
+
+    return string.match(rockspec, '^(%g+)%-scm%-1%.rockspec$')
```

Comment:
hm...

Why "scm", why "scm-1"? E.g in watchdog we use "scm-2" (https://github.com/tarantool/watchdog/blob/master/watchdog-scm-2.rockspec)

Reply:
I changed the name detecting approach.
I don't use rockspec filename anymore, I use `package` value from rockspec.